tests/Tests/Aggregations/Bucket/DateHistogram/DateHistogramAggregationUsageTests.cs

```diff
@@ -209,4 +210,82 @@ protected override void ExpectResponse(ISearchResponse<Project> response)
 			}
 		}
 	}
+
+	// hide
+	[SkipVersion("<7.10.0", "hard_bounds introduced in 7.10.0")]
+	public class DateHistogramAggregationWithHardBoundsUsageTests : ProjectsOnlyAggregationUsageTestBase
+	{
+		private readonly DateTime _hardBoundsMinimum;
+		private readonly DateTime _hardBoundsMaximum;
+		
+		public DateHistogramAggregationWithHardBoundsUsageTests(ReadOnlyCluster i, EndpointUsage usage) : base(i, usage)
+		{
+			// Note: If these tests are run against an existing node, and seeding is not forced, it's possible the
+			// dates used will not appear in the index and result in no buckets being returned. The test will still
+			// pass if this is the case. For best results locally, force a reseed. This is not an issue in CI.
+
+			var projects = Project.Projects.OrderBy(p => p.StartedOn).Skip(2).Take(5).ToArray();
+
+			_hardBoundsMinimum = DateTime.SpecifyKind(projects.First().StartedOn.Date, DateTimeKind.Unspecified);
```

Comment:
```suggestion
			_hardBoundsMinimum = DateTime.SpecifyKind(projects.Min(p=>p.StartedOn.Date), DateTimeKind.Unspecified);
```

Why is `DateTimeKind.Unspecified` needed?

Reply:
Done!

`DateTimeKind.Unspecified` is an interesting one. Without that we get back a server error:

```
"type" : "parse_exception",
        "reason" : "failed to parse date field [2019-11-27T00:00:00+00:00] with format [yyyy-MM-dd'T'HH:mm:ss]: [failed to parse date field [2019-11-27T00:00:00+00:00] with format [yyyy-MM-dd'T'HH:mm:ss]]",
```

Dates from `Project.Projects` are created a local kind. Later,`DateMath.ToString` handling of date formatting adds a suffix which ES is not happy about - https://github.com/elastic/elasticsearch-net/blob/master/src/Nest/CommonOptions/DateMath/DateMath.cs#L132

Would love to avoid this if there's a nicer way! 😄

Reply:
I _think_ Elasticsearch might be using the format string passed in the test in the `Format` property, which doesn't include an offset, to parse the date into a long bounds:

https://github.com/elastic/elasticsearch/blob/0af410ad0b686af21f9a5cff25ceff2b56bf4863/server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregationBuilder.java#L442-L446

and

https://github.com/elastic/elasticsearch/blob/4aaffc6a3deef28e8dedd3d9211951e3db0e2d8e/server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/LongBounds.java#L155-L171

I think that format is also used to format dates for the bucket key strings, in `ExpectResponse()`

Reply:
++ for this test it might sufficient to remove `format`, we might want to either add some `xmldocs` on `Format()` that its an expert option and might break the parsing. 

Sadly there is no way for us to carry over the chosen formatting in the response parsing.

Reply:
I think I ran into other issues with no formatting set at all. Let me double check that.

Reply:
Seems I can't replicate an issue right now. Running tests locally and if that all looks good I'll push a change up.

Reply:
Ok! Hopefully the latest change resolves this and works in CI. Everything ran correctly in local dev.